app/models/manageiq/providers/kubernetes/container_manager/event_catcher_mixin.rb

```diff
@@ -57,8 +64,15 @@ def process_event(event)
     when 'Node'
       event_data[:container_node_name] = event_data[:name]
     when 'Pod'
+      if event_data.key?(:fieldpath)
+        if event_data[:fieldpath].include? 'spec.containers'
+          event_data[:container_name] = event_data[:fieldpath][16..-2]
+          event_data[:kind] = 'Container'
+        end
```

Comment:
> This is even bugged I suppose, what do you do if event_data.key?(:fieldpath) is present (always) and include? 'spec.containers' is false? Is the event lost? (pretty much every pod event)

fieldpath key isn't always present. from types.go:
"Optional. If referring to a piece of an object instead of an entire object, this string should contain information to identify the sub-object ... This syntax is chosen only to have some well-defined way of referencing a part of an object."

So my thought was - if this field exists we want to extract the relevant container info from it (in the future we can add more options) and if it doesn't include container than we aren't handling this event for now. If I'm not mistaken the pod events we support are still caught.